federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/membership/group/GroupLDAPStorageMapper.java

```diff
@@ -361,22 +363,42 @@ private void updateAttributesOfKCGroup(GroupModel kcGroup, LDAPObject ldapGroup)
     }
 
 
-    protected GroupModel findKcGroupByLDAPGroup(RealmModel realm, LDAPObject ldapGroup) {
+    protected GroupModel findKcGroupByLDAPGroup(RealmModel realm, GroupModel parent, LDAPObject ldapGroup) {
         String groupNameAttr = config.getGroupNameLdapAttribute();
         String groupName = ldapGroup.getAttributeAsString(groupNameAttr);
 
         if (config.isPreserveGroupsInheritance()) {
             // Override if better effectivity or different algorithm is needed
-            return getAllKcGroups(realm)
+            return getAllKcGroups(realm, parent)
                     .filter(group -> Objects.equals(group.getName(), groupName)).findFirst().orElse(null);
         } else {
             // Without preserved inheritance, it's always at groups path
-            return KeycloakModelUtils.findGroupByPath(realm, getKcGroupPathFromLDAPGroupName(groupName));
+            return realm.getGroupByName(groupName, parent);
+        }
+    }
+    // TODO how this will work with group names with a / without hitting the db everytime
+    private GroupModel findGroupByPath(RealmModel realm, String path) {
```

Comment:
Instead of introducing this method, will it be better if the `KeycloakModelUtils.findGroupByPath` is changed this way by default? If it is better in your environment, it will be most likely for other environments as well.

IMO it does not makes sense that just GroupLDAPMapper uses this "optimized" version when all other Keycloak places use the "default" version, which is most likely worse for performance.

Reply:
That was my wish as well. But there is a handling of groups with / in the name in `KeycloakModelUtils.findGroupByPath` Any good idea how to solve this?

Reply:
This code is for the groupPath in the GroupLDAPStorageMapper.
`.property().name(GroupMapperConfig.LDAP_GROUPS_PATH)
                .label("Groups Path")
                .helpText("Keycloak group path the LDAP groups are added to. For example if value '/Applications/App1' is used, "
                		+ "then LDAP groups will be available in Keycloak under group 'App1', which is child of top level group 'Applications'. "
                		+ "The default value is '/' so LDAP groups will be mapped to the Keycloak groups at the top level. "
                		+ "The configured group path must already exists in the Keycloak when creating this mapper.")`

Reply:
@mhajas ok we can start with your approach. It might be good enough to extend the validation in GroupLDAPStorageMapper that a group with a '/' in the Name is not good for performance

Reply:
I like documenting this behaviour in the tooltip/docs better than refusing configs with a slash in the name. As I said, there may be a use case where someone wants to use a group with a slash in its name. If the validation refuses such config, they cannot do it. The only thing we can do with this is to create a new discussion/issue where we discuss how to get rid of groups with `/` in the name completely from the whole Keycloak. We can't change it only in one part of the code. @sguilhen Any thoughts on this?

Reply:
@mhajas I will try to support '/' in groupNames. Give me a few days. I'm a little bit busy with other tasks

Reply:
@mhajas I agree with your view that we should not refuse groups with slash in the name as that is currently allowed. As you pointed out, we can discuss whether this makes sense or not in a separate issue, but for the scope of this PR this is a behavior I would not change.

Reply:
@mhajas support for '/' in groupNames added. Can you give me some hints where and which kind of tests I should create. I'm not familar with the testing infrastructure of keycloak. There is no real test of KeycloakModelUtils.findGroupByPath beside testsuite/federation/ldap

Reply:
Thank you for the changes @bohmber! 

Regarding tests, we have some tests where we use `findGroupByPath`. For example, https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPGroupMapperSyncTest.java

The changes you added can be tested without LDAP. Not sure if we need any additional LDAP tests as ^LDAPGroupMapperSyncTest should cover the changes you made to the group mapper. In my opinion, what we need is the following:
1. Tests for the newly added method to GroupLookupProvider (`GroupModel getGroupByName(RealmModel realm, GroupModel parent, String name)`). We can create some groups, ideally with conflicting names under more parents and make sure the method returns the correct one. Another test can test the invalidation: call `getGroupByName` which should cache the result, then try to remove the group and make sure the follow-up call will return `null`. This can be added here: https://github.com/keycloak/keycloak/blob/main/testsuite/model/src/test/java/org/keycloak/testsuite/model/group/GroupModelTest.java see https://github.com/keycloak/keycloak/blob/main/testsuite/model/README.md on how to run
2. Then we should also add some tests for the new `findGroupByPath` method. It can be added to the same test class as 1. Let me know if you need any help, we can work together on that.